docs/examples/2Dwavefunc_animated.py

```diff
@@ -0,0 +1,193 @@
+"""
+===============================================
+Animating a time-varying 2D wave function
+===============================================
+
+This is a simple demonstration of how one can visualize
+time-varying 2D wave functions using FURY.
+
+Some other interesting functions you can try plotting -
+#z = (x**2 - y**2)/(x**2 + y**2)**0.5*np.cos(np.pi*x-rate*time)
+#z = np.sin(x**2)*np.cos(y**2-rate*incre)#*np.sin(rate*time)
+#z = np.sin(x**2-y**2)*np.sin(time)
+
+"""
+
+###############################################################################
+# Importing necessary modules
+
+from fury import window, actor, ui, utils, colormap
+from vtk.util import numpy_support
+import numpy as np
+import itertools
+
+
+###############################################################################
+# The following function is used to create and/or update the coordinates of the
+# points which are being used to plot the wave.
+# Kindly note that only the z coordinate is being modified with time as only
+# the z coordinate is a function of time.
+# When 'animating' is set to False, it returns colors and vertices that can be
+# used to generate a wave.
+# When 'animating' is set to True, it returns the updated colors and vertices
+# for the actor being used to generate the wave. Thus, 'animating' is set to
+# True when updating the coordinates inside the timer_callback function.
+
+def update_coordinates(lower_xbound=-1, upper_xbound=1, lower_ybound=-1,
+                       upper_ybound=1, npoints=100, time=0, incre=0,
+                       cmap_name='viridis', animating=True):
+
+    # creating the points which will be used to generate the surface
+    x = np.linspace(lower_xbound, upper_xbound, npoints)
+    y = np.linspace(lower_ybound, upper_ybound, npoints)
+    x, y = np.meshgrid(x, y)
+    x = x.reshape(-1)
+    y = y.reshape(-1)
+
+    # manipulate rate to change the speed of the waves; (default = 1.5)
+    rate = 1.5
+
+    # Z is the function F i.e. F(x, y, t)
+    z = 0.24*np.sin(4*np.pi*x)*np.cos(4*np.pi*y-rate*time)
+    xyz = np.vstack([x, y, z]).T
+
+    # creating the colormap
+    v = np.copy(z)
+    v /= np.max(np.abs(v), axis=0)
+    colors = colormap.create_colormap(v, name=cmap_name)
+    if animating:
+        temp_surface = actor.surface(xyz, colors=colors)
```

Comment:
Creating a temporary surface every time while updating the vertice colors is something that takes a lot of time, is there a better way to update the vertice colors?

Reply:
Resolved it by directly updating colors of the actor without making a temporary surface.